pkg/codec.go

```diff
@@ -362,3 +391,21 @@ func processAttachments(decoder iface.Decoder, message remote.Message) error {
 
 	return fmt.Errorf("unsupported attachments: %v", attachmentsRaw)
 }
+
+func getTypes(data interface{}) (string, error) {
```

Comment:
It think it's better moved to pkg/dubbo, and we can maintain a cache with key = reflect.ValueOf(data)